aspnet/fundamentals/file-providers.rst

```diff
@@ -1,8 +1,149 @@
+
+To create a ``PhysicalFileProvider``, simply instantiate it, passing it a physcial path. You can then iterate through its directory contents or get a specific file's information by providing a subpath.
+
+.. code-block:: c#
+
+  IFileProvider provider = new PhysicalFileProvider(applicationRoot);
+  IDirectoryContents contents = provider.GetDirectoryContents(""); // the applicationRoot contents
+  IFileInfo fileInfo = provider.GetFileInfo("wwwroot/js/site.js"); // a file under applicationRoot
+
+To request a provider from a controller, specify it in the controller's constructor and assign it to a local field. Use the local instance from your action methods:
+
+.. literalinclude:: file-providers/sample/src/FileProviderSample/Controllers/HomeController.cs
+  :lines: 7-21
+  :emphasize-lines: 6
+  :dedent: 4
+
+Then, create the provider in the app's ``Startup`` class:
+
+.. literalinclude:: file-providers/sample/src/FileProviderSample/Startup.cs
+  :lines: 11-15, 23-24, 29-34, 38, 41
+  :emphasize-lines: 13-14
+  :dedent: 4
+
+In the *Index.cshtml* view, iterate through the ``IDirectoryContents`` provided:
+
+.. literalinclude:: file-providers/sample/src/FileProviderSample/Views/Home/Index.cshtml
+  :emphasize-lines: 2,7,9,11,15
+
+The result:
+
+.. image:: /fundamentals/file-providers/_static/physical-directory-listing.png
+
+EmbeddedFileProvider
+^^^^^^^^^^^^^^^^^^^^
+
+The ``EmbeddedFileProvider`` is used to access files embedded in assemblies. In .NET Core, you embed files in an assembly by specifying them in ``buildOptions`` in the *project.json* file:
+
+.. literalinclude:: file-providers/sample/src/FileProviderSample/project.json
+  :lines: 42-49
+  :emphasize-lines: 4-7
+
+You can use `globbing patterns`_ when specifying files to embed in the assembly.
+
+When creating an ``EmbeddedFileProvider``, pass the assembly it will read to its constructor.
+
+.. code-block:: c#
+
+  var embeddedProvider = new EmbeddedFileProvider(Assembly.GetEntryAssembly());
+
+The snippet above demonstrates how to create an ``EmbeddedFileProvider`` with access to the currently executing assembly.
+
+Updating the sample app to use an ``EmbeddedFileProvider`` results in the following output:
+
+.. image:: /fundamentals/file-providers/_static/embedded-directory-listing.png
+
+.. note:: Embedded resources do not expose directories. Rather, the path to the resource is embedded in its filename using ``.`` separators.
```

Comment:
Might be worth noting that `.GetDirectoryContents` will return resources under the base namespace given to the ctor `public EmbeddedFileProvider(Assembly assembly, string baseNamespace)`